test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresUnreferencedCodeCapability.cs

```diff
@@ -234,6 +235,22 @@ static void TestStaticCctorRequiresUnreferencedCode ()
 			_ = new StaticCtor ();
 		}
 
+		class StaticCtorTriggeredByFieldAccess
+		{
+			[RequiresUnreferencedCode ("Message for --StaticCtorTriggeredByFieldAccess.Cctor--")]
+			static StaticCtorTriggeredByFieldAccess ()
+			{
+			}
+
+			public static int field = 0;
```

Comment:
I think it'd be better not to use static fields to check that types without `beforefieldinit` modifiers work as expected.

Reply:
I think moving the field initialization into the static `.cctor` should be enough, right?

Reply:
I think the test coverage should be better. Does the code print the warning in this case?

```c#
new C ().Foo ();

class C
{
	[RequiresUnreferencedCode ("Message")]
    static C ()
    {
    }
    
    public void Foo ()
    {
    }
}
```

Reply:
Also, AFAIK Roslyn won't make the type beforefieldinit in any of the above cases - the explicit static ctor in C# doesn't have the relaxed semantics allowed by beforefieldinit. To test with beforefieldinit you need to avoid writing a static ctor and let Roslyn generate the `.cctor` from the static field initializers:
```csharp
class C {
    static int field = RUC();
    [RequiresUnreferencedCode]
    static int RUC() => 0;
}
```